File: Leetcode/9.Palindrome Number/C++/main.cpp
#include <bits/stdc++.h>
using namespace std;
//start of solution
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)
            return false;
        long long reverse = 0,temp = x;
        while(temp){
            reverse += temp%10;
            reverse*=10;
            temp/=10;
        }
        reverse/=10;
        return reverse == x;
    }
};
//end of solution

int main(){
    Solution sol1, sol2;
    if(sol1.isPalindrome(12))
        cout<<"yes\n";
    else 
        cout<<"no\n";

    if(sol1.isPalindrome(121))
        cout<<"yes\n";
    else 
        cout<<"no\n";
    return 0;
}

File: Leetcode/9.Palindrome Number/Python/main.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        div =1
        
        while x >= div *10:
            div*=10
            
        while x:
            right = x % 10
            left = x // div
            if left != right:
                return False
            x = (x%div)//10
            div = div /100
        return True

sol1,sol2 = Solution(),Solution()
print(sol1.isPalindrome(12))
print(sol2.isPalindrome(121))